woocommerce/woocommerce-gutenberg-products-block
# Rename onCheckoutBeforeProcessing to onCheckoutValidationBeforeProcessing

This commit also adds a deprecation warning when using onCheckoutBeforeProcessing.
See the issue below for details.


Fixes #3984





### How to test the changes in this Pull Request:
The renaming part could be tested by smoke testing the Checkout Block:

1. Add something to checkout
2. Submit without a required field
3. Observe validation happens
4. Fix validation and place order

To see the warning you have to call yourself the old `onCheckoutBeforeProcessing`. What I did:

1. I added this component to [Order Summary](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/assets/js/base/components/cart-checkout/order-summary/index.js) and included it as a child.
2. I went to the Checkout block and tried to submit it without some required fields.
3. I checked the console for the deprecation warning
4. I then fixed the warnings and  successfully submitted the order



```js
const Component = () => {
	const { onCheckoutBeforeProcessing } = useCheckoutContext();
	useEffect( () => {
		const unsubscribe = onCheckoutBeforeProcessing( () => {
			console.log( 'My custom validation callback' );
			return true;
		} );
		return unsubscribe;
	}, [ onCheckoutBeforeProcessing ] );
	return null;
};
```

One thing that could also help with testing is checking the registered observers by adding `console.log(observersByType)` in [emitters.ts](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/assets/js/base/context/shared/event-emit/emitters.ts#L26). This is logged after submitting with missing inputs.



### Changelog

> Rename onCheckoutBeforeProcessing to onCheckoutValidationBeforeProcessing.

### Dev note

> `onCheckoutBeforeProcessing` has been renamed to `onCheckoutValidationBeforeProcessing` to more clearly articulate its coupling to the validation happening when the Checkout form is submitted. `onCheckoutBeforeProcessing` will be deprecated in the future.
